test/integration/targets/docker-registry/tasks/main.yml

```diff
@@ -0,0 +1,139 @@
+---
+- name: Create random name prefix and test registry name
+  set_fact:
+    name_prefix: "{{ 'ansible-test-%0x' % ((2**32) | random) }}"
+    registry_name: "{{ 'ansible-test-registry-%0x' % ((2**32) | random) }}"
+    nginx_name: "{{ 'ansible-test-registry-frontend-%0x' % ((2**32) | random) }}"
+- name: Create image and container list
+  set_fact:
+    inames: []
+    cnames:
+    - "{{ registry_name }}"
+    - "{{ nginx_name }}"
+    vnames:
+    - "{{ nginx_name }}"
+
+- debug:
+    msg: "Using name prefix {{ name_prefix }} and test registry name {{ registry_name }}"
+
+- block:
+  - name: Start test registry
+    docker_container:
+      name: "{{ registry_name }}"
+      image: registry:2.6.1
+      ports: 5000
+    register: registry_container
+
+  - name: Get registry URL
+    set_fact:
+      registry_address: "localhost:{{ registry_container.container.NetworkSettings.Ports['5000/tcp'].0.HostPort }}"
+
+  - name: Start nginx frontend for registry
+    docker_volume:
+      name: "{{ nginx_name }}"
+      state: present
+
+  - name: Create container for nginx frontend for registry
+    docker_container:
+      state: stopped
+      name: "{{ nginx_name }}"
+      image: nginx:alpine
+      ports: 5000
+      links:
+        - "{{ registry_name }}:real-registry"
+      volumes:
+        - "{{ nginx_name }}:/etc/nginx/"
+    register: nginx_container
+
+  - name: Copy static files into volume
+    command: docker cp {{ role_path }}/files/{{ item }} {{ nginx_name }}:/etc/nginx/{{ item }}
+    loop:
+      - "nginx.conf"
+      - "nginx.htpasswd"
+
+  - name: Create private key for frontend certificate
```

Comment:
Good question. The advantage of not including them is that they won't expire at some random point of time. Also, ECC private keys are quick to generate, so regenerating them doesn't cost too much time. Finally, if we ever want to make certification validation work (for the changes in https://github.com/felixfontein/ansible/commit/df23b5991d5e03dbfbd87e578f2e419928e8f73b), we need to regenerate a cert every time since the IP isn't fixed.

So for now I'd keep the regeneration.